server/src/test/java/org/opensearch/snapshots/SnapshotUtilsTests.java

```diff
@@ -85,4 +93,48 @@ private void assertIndexNameFiltering(String[] indices, String[] filter, Indices
         List<String> actual = SnapshotUtils.filterIndices(indicesList, filter, indicesOptions);
         assertThat(actual, containsInAnyOrder(expected));
     }
+
+    public void testValidateSnapshotsBackingAnyIndex() {
+        final String repoName = "test-repo";
+        final SnapshotId snapshotId1 = new SnapshotId("testSnapshot1", "uuid1");
+        final SnapshotId snapshotId2 = new SnapshotId("testSnapshot2", "uuid2");
+        ClusterState clusterState = getClusterState(snapshotId1, repoName);
+        try {
+            SnapshotUtils.validateSnapshotsBackingAnyIndex(clusterState, List.of(snapshotId2), repoName);
+        } catch (Exception e) {
+            fail("Should not have thrown any exception");
```

Comment:
No need for the try-catch. The exception will result in a test failure and will contain specific details about why the validation failed.

Reply:
I am not sure I get it. I am using the try block to test the happy path and catch block for unhappy path. Since the particular test should never follow the unhappy path, it should never come into the catch block. If it executes the catch block, we fail the test.

Could you please elaborate your suggestion?

Reply:
Replace

```
try {
    SnapshotUtils.validateSnapshotsBackingAnyIndex(clusterState, List.of(snapshotId2), repoName);
} catch (Exception e) {
    fail("Should not have thrown any exception");
}
```

with

```
SnapshotUtils.validateSnapshotsBackingAnyIndex(clusterState, List.of(snapshotId2), repoName);
```

All fail() does is throw an AssertionError with the provided message. In this case it will end up hiding the details of the actual exception that caused the failure. It's better just to let _that_ exception fail the test in the case of a failure.

Reply:
Got it!